examples/memcached.md

```diff
@@ -0,0 +1,197 @@
+# memcached
+
+As shown in [this blog
+post](https://memcached.org/blog/persistent-memory/), memcached can
+efficiently utilize PMEM as replacement for DRAM. This makes it
+possible to increase cache sizes and/or reduce costs. The memcached
+maintainers make container images available which support this
+feature. This example shows how to take advantage of PMEM with
+memcached running on Kubernetes.
+
+## PMEM as DRAM replacement
+
+Memcached can map PMEM into its address space and then store the bulk
+of the cache there. The advantage is higher capacity and (depending on
+pricing) less costs.
+
+The cache could be stored in a persistent volume and reused after a
+restart (see next section), but when that isn't the goal, a
+[deployment with one CSI ephemeral inline
+volume](../deploy/kustomize/memcached/ephemeral/memcached-ephemeral.yaml)
+per memcached pod is very simple:
+- A Kubernetes deployment manages the memcached instance(s).
+- In the pod template, one additional PMEM volume of a certain size is
+  requested.
+- An init container prepares that volume for use by memcached:
+  - It determines how large the actual file can be that memcached will use.
+  - It changes volume ownership so that memcached can run as non-root
+    process.
+- A wrapper script adds the `--memory-file` and `--memory-limit` parameters
+  when invoking memcached, which enables the use of PMEM.
+
+For this to work, the PMEM-CSI driver must have been deployed with
+`pmem-csi.intel.com` as driver name. No storage classes are
+needed. That name and other parameters in the deployment can be
+modified with
+[`kustomize`](https://github.com/kubernetes-sigs/kustomize). Here's
+how one can change the namespace, volume size or add additional
+command line parameters:
+
+```
+$ mkdir -p my-memcached-deployment
```

Comment:
I'd rather change this so that the example works without having to check out code. The current ambiguity comes from me not quite knowing how to do that properly and thus ignoring the problem :grimacing: 

It can be done by replacing all file references with URLs. There are just two issues with that:
- the instructions will only work after merging (only a problem for reviewers, can be worked around)
- ideally, we want the versioned documentation for a release to refer to the versioned files, because otherwise we may end up with https://intel.github.io/pmem-csi/0.7/README.html pointing to files that no longer exist on the master branch (no solution yet)

It also would be nice if the URLs were actual URLs in HTML rendering, but that conflicts with putting them into code blocks.

Reply:
I got the following script from Graham to help reviewers validate unmerged code. I even used it to test these files out. You do a `git clone` of the repo, change into the directory, and then run the script with the pull number and it checks out the files for you.  In this case I would `./pullpr.sh 633`

```
$ cat pullpr.sh
#!/bin/bash
# Do a git clone of the repository in question
# Change to root directory of repo
# Run ./pullpr.sh 445   where 445 is the PR number

if [ $# -ne 1 ]; then
        echo "Require one argument (PR number)"
        exit 1
fi

pr=${1}
pr_branch=PR_${pr}

echo "Fetching PR ${pr} to branch ${pr_branch}"
git fetch origin "refs/pull/$pr/head:$pr_branch"
git checkout $pr_branch
git branch
```

Reply:
> I got the following script from Graham to help reviewers validate unmerged code.

There's also `hub checkout <PR URL>`. But as the goal is to make this example and the deployment files usable without people checking out code, it's better to also test the PR that way.